inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp

```diff
@@ -1261,7 +1263,8 @@ void MKLDNNGraphOptimizer::FuseEltwiseAndSimple(MKLDNNGraph &graph) {
     auto& graphNodes = graph.GetNodes();
 
     auto isSutableParentNode = [](MKLDNNNodePtr node) {
-        return node->getType() == Eltwise && node->getChildEdges().size() == 1;
+        return node->getType() == Eltwise && node->getChildEdges().size() == 1 &&
+               node->getParentEdgesAtPort(0).front()->getDims().ndims() <= MAX_ELTWISE_DIM_RANK;
```

Comment:
You perform this check inside EltwiseNode::canFuse. Why do we need it here?

Reply:
I haven't noticed method calling `canFuse` in this check. Removed, thank you!